File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/ViewCardDemo/ViewCardDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include "shellapi.hpp"
#pragma hdrstop

#include "ViewCardDemoMain.h"
#include "ViewCardDemoData.h"
#include "AboutDemoForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxDBData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxGrid"
#pragma link "cxGridCustomTableView"
#pragma link "cxGridCustomView"
#pragma link "cxGridDBTableView"
#pragma link "cxGridLevel"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxButtons"
#pragma link "cxCheckBox"
#pragma link "cxContainer"
#pragma link "cxGridCardView"
#pragma link "cxGridDBCardView"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxMaskEdit"
#pragma link "cxSpinEdit"
#pragma link "cxTextEdit"
#pragma link "cxLookAndFeels"
#pragma link "cxDataStorage"
#pragma link "cxDBLookupComboBox"
#pragma link "cxGridCustomLayoutView"
#pragma link "cxHyperLinkEdit"
#pragma link "cxImageComboBox"
#pragma link "cxMemo"
#pragma resource "*.dfm"
TViewCardDemoMainForm *ViewCardDemoMainForm;
//---------------------------------------------------------------------------
__fastcall TViewCardDemoMainForm::TViewCardDemoMainForm(TComponent* Owner)
  : TfmBaseForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TViewCardDemoMainForm::miShowEmptyRowsClick(TObject *Sender)
{
  cvPersons->OptionsView->EmptyRows = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TViewCardDemoMainForm::miFilteringClick(TObject *Sender)
{
  cvPersons->OptionsCustomize->RowFiltering = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TViewCardDemoMainForm::miExpandingCollapsingClick(TObject *Sender)
{
  cvPersons->OptionsCustomize->CardExpanding = ((TMenuItem*)Sender)->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TViewCardDemoMainForm::miRowsCustomizationClick(TObject *Sender)
{
  cvPersons->Controller->Customization = true;
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/ViewCardDemo/ViewCardDemoMain.pas
unit ViewCardDemoMain;

interface

uses
  Windows, Forms, Messages, SysUtils, Classes, ActnList, ImgList, Controls, Menus,
  StdCtrls, cxButtons, cxCheckBox, cxContainer, cxEdit, cxTextEdit,
  cxMaskEdit, cxSpinEdit, ExtCtrls, cxGridLevel, cxGridCustomTableView,
  cxGridCardView, cxGridDBCardView, cxClasses, cxControls,
  cxGridCustomView, cxGrid, ComCtrls, cxStyles, cxCustomData, cxGraphics,
  cxFilter, cxData, DB, cxDBData, cxDataStorage, cxLookAndFeelPainters,
  cxLookAndFeels, cxHyperLinkEdit, cxImageComboBox, cxDBLookupComboBox,
  cxMemo, cxGridCustomLayoutView, BaseForm, cxGridTableView;

type
  TViewCardDemoMainForm = class(TfmBaseForm)
    miOptions: TMenuItem;
    lvPersons: TcxGridLevel;
    cxGrid: TcxGrid;
    cvPersons: TcxGridDBCardView;
    cvPersonsID: TcxGridDBCardViewRow;
    cvPersonsFIRSTNAME: TcxGridDBCardViewRow;
    cvPersonsSECONDNAME: TcxGridDBCardViewRow;
    cvPersonsGENDER: TcxGridDBCardViewRow;
    cvPersonsBIRTHNAME: TcxGridDBCardViewRow;
    cvPersonsDATEOFBIRTH: TcxGridDBCardViewRow;
    cvPersonsBIRTHCOUNTRY: TcxGridDBCardViewRow;
    cvPersonsLOCATIONOFBIRTH: TcxGridDBCardViewRow;
    cvPersonsBIOGRAPHY: TcxGridDBCardViewRow;
    cvPersonsNICKNAME: TcxGridDBCardViewRow;
    cvPersonsFullname: TcxGridDBCardViewRow;
    cvPersonsHOMEPAGE: TcxGridDBCardViewRow;
    miShowEmptyRows: TMenuItem;
    miFiltering: TMenuItem;
    miExpandingCollapsing: TMenuItem;
    N1: TMenuItem;
    miRowsCustomization: TMenuItem;
    procedure miShowEmptyRowsClick(Sender: TObject);
    procedure miFilteringClick(Sender: TObject);
    procedure miExpandingCollapsingClick(Sender: TObject);
    procedure miRowsCustomizationClick(Sender: TObject);
  end;

var
  ViewCardDemoMainForm: TViewCardDemoMainForm;

implementation

{$R *.dfm}

uses
  ViewCardDemoData, Dialogs, AboutDemoForm;

procedure TViewCardDemoMainForm.miShowEmptyRowsClick(Sender: TObject);
begin
  cvPersons.OptionsView.EmptyRows := GetMenuItemChecked(Sender);
end;

procedure TViewCardDemoMainForm.miFilteringClick(Sender: TObject);
begin
  cvPersons.OptionsCustomize.RowFiltering := GetMenuItemChecked(Sender);
end;

procedure TViewCardDemoMainForm.miExpandingCollapsingClick(Sender: TObject);
begin
  cvPersons.OptionsCustomize.CardExpanding := GetMenuItemChecked(Sender);
end;

procedure TViewCardDemoMainForm.miRowsCustomizationClick(Sender: TObject);
begin
  cvPersons.Controller.Customization := True;
end;

end.
